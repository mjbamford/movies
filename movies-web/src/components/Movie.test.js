import React from 'react';
import Movie from './Movie'
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

describe('Movie', () => {
    const attrs = { _id: 666, title: 'Title 666', yearReleased: 1990 }
    it('renders the movie correctly', () => {
        const wrapper = shallow(<Movie {...attrs} />)
        expect(wrapper.contains('Title 666')).toEqual(true)
    })

    it('renders the snapshot', () => {
        const movie = renderer.create(
            <MemoryRouter>
                <Movie {...attrs} />
            </MemoryRouter>
        )
        expect(movie.toJSON()).toMatchSnapshot()
    })
})
