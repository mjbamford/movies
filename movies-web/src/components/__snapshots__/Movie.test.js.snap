// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders the snapshot 1`] = `
<div>
  <a
    href="/movies/666"
    onClick={[Function]}
  >
    Title 666
  </a>
  <span>
    1990
  </span>
  <span>
     
  </span>
  <small>
    <a
      href="/movies/666/edit"
      onClick={[Function]}
    >
      Edit
    </a>
  </small>
  <ul />
</div>
`;
